platform/feature-api/api/src/pod-api/api.ts

```diff
@@ -95,6 +122,21 @@ export interface PolyIn {
      * @returns a Promise that will be resolved to a boolean.
      */
     has(...quads: RDF.Quad[]): Promise<boolean>;
+
+    /**
+     * SPARQL query (https://www.w3.org/TR/sparql11-query/) against the rdf database
+     *
+     * @param query SPARQL query string
+     * @returns a promise that resolves to a queryResult
+     */
+    query(query: string): Promise<QueryResult>;
```

Comment:
Absolutely! There needs to go way more work into the permission model with the graphs and such (I focused on the default graph here)

Reply:
I don't understand the parse and validate step though. Oxi parses the queries and we can return an error type for incorrect query from there

Reply:
Just parsing the query for correctness, before submitting it to the actual query engine. That would allow us to extract some information, at least know if it's a regular or update query, too.

Reply:
Would you want to have one interface for querying, which knows what it should call?

Reply:
Not totally sure about that. `if`s on the arguments are never a good practice. Maybe at some point, I don't know. Repeating the two functions at all levels becomes a bit tiring. Maybe at the highest level, since it might make things easier for feature developers. At any rate, there's bound to be a high-level API; you mentioned you were checkout out templates that might hide the complexity of SPARQL querying, but even at that level "query" or "update" as separate functions might make sense.

Reply:
Yeah I agree, we will try to hide it all with poly-rdf. I agree with keeping them for now mostly due to the different SPARQL specifications https://www.w3.org/TR/sparql11-query/ and https://www.w3.org/TR/sparql11-update/